### PR TITLE
Save REPL history to file .dotty_history in home directory

### LINKED ISSUE #5142: REPL history is not saved between sessions
JLine has support for saving and restoring the history but it doesn't look like we use this in the dotty repl currently.

### PR DESCRIPTION
**Pull request by nrjais**
Fixes #5142

### CODE REVIEW

File: compiler/src/dotty/tools/repl/JLineTerminal.scala [hunk 1]
```diff
@@ -47,12 +47,15 @@ final class JLineTerminal(needsTerminal: Boolean) extends java.io.Closeable {
...
-    val lineReader = LineReaderBuilder.builder()
+    val userHome = System.getProperty("user.home")
+    val lineReader = LineReaderBuilder
+      .builder()
       .terminal(terminal)
       .history(history)
       .completer(completer)
       .highlighter(new Highlighter)
       .parser(new Parser)
+      .variable(HISTORY_FILE, s"$userHome/.dotty_history") // Save history to file
```
Comment (smarter) [inline compiler/src/dotty/tools/repl/JLineTerminal.scala]: Does this work on Windows or do we need File.pathSeparator instead of / here too ?

Comment (allanrenucci) [reply compiler/src/dotty/tools/repl/JLineTerminal.scala]: This goes through `java.nio.file.Paths`. I believe it handles path separator correctly

Comment (smarter) [reply compiler/src/dotty/tools/repl/JLineTerminal.scala]: Ok. Note for later: To be perfect we probably shouldn't write files directly in the home directory, on Linux you should use $HOME/.config/nameofyourapplication/ and on Windows there's some other folder, but this is hard to handle correctly. There's https://github.com/soc/directories-jvm that is designed to do that for you, but it doesn't seem to have a stable ABI so we'd have to embed it in our sources.

Comment (milessabin) [reply compiler/src/dotty/tools/repl/JLineTerminal.scala]: I think `$HOME/.dotty_history` is the right place on Linux. A history file isn't a configuration file, so it doesn't belong under `$HOME/.config`. Also cp. `$HOME/.scala_history`, `$HOME/.histfile`, `$HOME/.bash_history` and many, many more ...

For configuration files on Linux, if you're going to bother with the [XDG spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), you should first check if `$XDG_CONFIG_HOME` is set and only default to `$HOME/.config` if it isn't. But bear in mind that standards schmandards and lots of configuration stuff still quite happily lives in dotfiles directly in `$HOME`.

**Pull request merged**